remote/remote_test.go

```diff
@@ -501,6 +503,77 @@ func testImage(t *testing.T, when spec.G, it spec.S) {
 				})
 			})
 		})
+
+		when("#RetryLogic", func() {
+			var server *httptest.Server
+			var repo string
+
+			when("Manifest API in registry returns status code 200", func() {
+				repo = "org/do-not-retry"
+				mockServer := h.NewMockServer(repo, http.StatusOK, 0)
+
+				it.Before(func() {
+					server = mockServer.Server()
+					u, err := url.Parse(server.URL)
+
+					h.AssertNil(t, err)
+
+					repoName = u.Hostname() + ":" + u.Port() + "/" + repo
+				})
+
+				it("Do not retry after status code 200", func() {
+					defer server.Close()
+					_, err := remote.NewImage(repoName, authn.DefaultKeychain, remote.WithPreviousImage(repoName))
+
+					h.AssertNil(t, err)
+					h.AssertEq(t, mockServer.ActualCount(), 1)
+				})
+			})
+
+			when("Manifest API in registry returns status code 404", func() {
+				repo = "org/retry-not-found"
+				mockServer := h.NewMockServer(repo, http.StatusNotFound, 2)
+
+				it.Before(func() {
+					server = mockServer.Server()
+					u, err := url.Parse(server.URL)
+
+					h.AssertNil(t, err)
+
+					repoName = u.Hostname() + ":" + u.Port() + "/" + repo
+				})
+
+				it("Retry after status code 404", func() {
+					defer server.Close()
+					_, err := remote.NewImage(repoName, authn.DefaultKeychain, remote.WithPreviousImage(repoName))
+
+					h.AssertNil(t, err)
+					h.AssertEq(t, mockServer.ActualCount(), 3)
```

Comment:
Same comment as https://github.com/buildpacks/imgutil/pull/111/files#r631156024.

Reply:
I put it in the same `remote_test.go` file because I need to access the remote package, is that ok?